tests/unit/test_crunch_cube.py

```diff
@@ -432,3 +392,413 @@ def test_inserted_inds(self, mock_inserted_hs_indices,
         # Assert indices are fetch with transforms
         actual = cc._inserted_dim_inds(Mock(), 0)
         mock_inserted_hs_indices.assert_called_once()
+
+    @patch('cr.cube.crunch_cube.CrunchCube.all_dimensions', [
+        Mock(type='categorical', is_selections=False),
+        Mock(type='categorical', is_selections=False),
+    ])
+    def test_adjust_axes_cat_x_cat(self):
```

Comment:
These adjust axes look really nice, and a great way to conceptualize what's going on.